### PR TITLE
feat: add image scanning and refactor Makefile

### LINKED ISSUE #255: Build multi-arch images for 1809, 1903 and 1909
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### LINKED ISSUE #311: Add image vulnerability scan as part of CI
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Adds image scan as part of CI
- Refactors `Makefile`
- Adds target for windows multi-arch image builds
- Adds log collection as part of CI to enable debugging

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #311 
fixes #255

**Please answer the following questions with yes/no**:

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Special Notes for Reviewers**:

### CODE REVIEW

File: .pipelines/templates/scan-images.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+steps:
+  - script: |
+      export REGISTRY="e2e"
+      export IMAGE_VERSION="test"
+      OUTPUT_TYPE=docker  make container
+      wget https://github.com/aquasecurity/trivy/releases/download/v0.11.0/trivy_0.11.0_Linux-64bit.tar.gz
```
Comment (sozercan) [inline .pipelines/templates/scan-images.yaml:6]: can you make trivy version a variable?

Comment (aramase) [reply .pipelines/templates/scan-images.yaml]: done!

File: .pipelines/templates/cleanup-images.yaml [hunk 2]
```diff
@@ -4,8 +4,11 @@ steps:
...
-      az acr repository delete --name $(REGISTRY_NAME) --image secrets-store-csi-driver-provider-azure:${IMAGE_VERSION}-windows-1809-amd64 -y || true
-      az acr repository delete --name $(REGISTRY_NAME) --image secrets-store-csi-driver-provider-azure:${IMAGE_VERSION} -y || true
+      az acr repository delete --name $(REGISTRY_NAME) --image k8s/csi/secrets-store/provider-azure:${IMAGE_VERSION}-linux-amd64 -y || true
+      az acr repository delete --name $(REGISTRY_NAME) --image k8s/csi/secrets-store/provider-azure:${IMAGE_VERSION}-windows-1809-amd64 -y || true
+      az acr repository delete --name $(REGISTRY_NAME) --image k8s/csi/secrets-store/provider-azure:${IMAGE_VERSION}-windows-1903-amd64 -y || true
+      az acr repository delete --name $(REGISTRY_NAME) --image k8s/csi/secrets-store/provider-azure:${IMAGE_VERSION}-windows-1909-amd64 -y || true
+      az acr repository delete --name $(REGISTRY_NAME) --image k8s/csi/secrets-store/provider-azure:${IMAGE_VERSION}-windows-2004-amd64 -y || true
+      az acr repository delete --name $(REGISTRY_NAME) --image k8s/csi/secrets-store/provider-azure:${IMAGE_VERSION} -y || true
```
Comment (ritazh) [inline .pipelines/templates/cleanup-images.yaml:12]: You might want to check for empty string for the image tag. Apparently an empty tag will result in deleting the whole repo.

File: Makefile [hunk 3]
```diff
@@ -42,71 +63,90 @@ lint: $(TOOLS_DIR)/golangci-lint $(TOOLS_DIR)/misspell
...
+
+.PHONY: container-windows
+container-windows: docker-buildx-builder
+	docker buildx build --no-cache --output=type=$(OUTPUT_TYPE) --platform="windows/amd64" \
+	 		-t $(IMAGE_TAG)-windows-$(OSVERSION)-$(ARCH) --build-arg OSVERSION=$(OSVERSION) -f windows.Dockerfile .
+
+.PHONY: docker-buildx-builder
+docker-buildx-builder:
+	@if ! DOCKER_CLI_EXPERIMENTAL=enabled docker buildx ls | grep -q container-builder; then\
+		DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name container-builder --use --driver-opt image=moby/buildkit:v0.7.2;\
```
Comment (ritazh) [inline Makefile:99]: lets make buildkit version a variable

File: .pipelines/templates/cleanup-images.yaml (hunk 2 above)
Comment (aramase) [reply .pipelines/templates/cleanup-images.yaml]: Good suggestion. Done!

File: Makefile (hunk 3 above)
Comment (aramase) [reply Makefile]: Done! Also updated to `buildkit:v0.8.1` which contains the fix for windows image builds.

**Review by ritazh — APPROVED** [review]

**Pull request merged**